### PR TITLE
chore: update to OpenSSL 3.0.0

### PR DESCRIPTION
**Pull request by mayeut**
The OpenSSL 1.1.1 series is still supported until 11th September 2023.
Maybe we should wait to upgrade to the 3.0 series ? Maybe not ?

### CODE REVIEW

**Review by henryiii — APPROVED** [review]
Fine by me.

**Comment by mayeut** [comment]
Let's put this one on hold until #193 gets merged to check a rebase with musllinux is still ok.

**Pull request merged**
